lib/src/store.dart

```diff
@@ -135,6 +135,7 @@ class Store extends Stream<Store> with Disposable {
   /// called until that future has resolved.
   ///
   /// If the `Store` has been disposed, this method throws a [StateError].
+  @Deprecated('4.0.0')
```

Comment:
Currently at 2.9.4 .. is this the right version to remove this?

Reply:
Our recommended pattern for deprecations is to just use `@deprecated` and include version info and explanation in the doc comment. Example:

```dart
/// An HTTP client acts as a single point from which many requests...
///
/// Deprecated: 2.6.4
/// To be removed: 3.0.0
///
/// This class has been renamed. Use `HttpClient` instead.
@deprecated
class Client {

}
```

https://dev.workiva.net/docs/platform/application-frameworks/front-end-dev-policies/wdesk-versioning/deprecation-strategies

Reply:
Oh, and to actually answer your question.. :) it would be 3.0.0